// log-tips/dist/log-tips.js
(function (global, factory) {
  typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports) :
  typeof define === 'function' && define.amd ? define(['exports'], factory) :
  (global = global || self, factory(global.warning = {}));
}(this, function (exports) { 'use strict';

  /**
   * Check if argument is an Number.
   *
   * @param {Object} value
   * @return {Boolean}
   */
  var isNum = function isNum(num) {
    return typeof num === 'number' && !isNaN(num);
  };
  /**
   * Check if argument is an Object.
   *
   * @param {Object} value
   * @return {Boolean}
   */


  var isObject = function isObject(value) {
    var type = Object.prototype.toString.call(value);
    return type === '[object Object]';
  };

  var isArray = Array.isArray.bind(Array);
  /**
   * Check if argument is a function.
   *
   * @param {Object} value
   * @return {Boolean}
   */

  var isFn = function isFn(o) {
    return typeof o === 'function';
  };

  var noop = function noop() {};
  /**
   * Check if argument is a HTML element.
   *
   * @param {Object} value
   * @return {Boolean}
   */


  var isNode = function isNode(value) {
    return value !== undefined && (value instanceof HTMLElement && value.nodeType === 1 || value.nodeType === 9);
  };
  /**
   * Check if argument is a list of HTML elements.
   *
   * @param {Object} value
   * @return {Boolean}
   */


  var isNodeList = function isNodeList(value) {
    var type = Object.prototype.toString.call(value);
    return value !== undefined && (type === '[object NodeList]' || type === '[object HTMLCollection]') && 'length' in value && (value.length === 0 || exports.node(value[0]));
  };
  /**
   * Check if argument is a SVG element.
   *
   * @param {Object} value
   * @return {Boolean}
   */


  var isSvg = function isSvg(value) {
    return value !== undefined && value instanceof SVGElement;
  };
  /**
   * Check if argument is a string.
   *
   * @param {Object} value
   * @return {Boolean}
   */


  var isString = function isString(value) {
    return typeof value === 'string' || value instanceof String;
  };

  /**
   * isEffect
   *
   * @param {mixed} condition  触发条件
   * @param {string} message 触发提示
   * @param {string} [type='error'] 触发类型
   * @returns
   */

  function isEffect(condition, message, type // log | error | warn
  ) {
    if (type === void 0) {
      type = 'error';
    }

    {
      // condition passed: do not log
      if (condition) {
        return;
      } // Condition not passed


      var text = type + ": " + message;

      try {
        console[type.toLowerCase()](text);
      } catch (x) {
        console.log(x);
      }
    }
  }

  exports.default = isEffect;
  exports.isNum = isNum;
  exports.isFn = isFn;
  exports.isNode = isNode;
  exports.isNodeList = isNodeList;
  exports.isObject = isObject;
  exports.isString = isString;
  exports.isArray = isArray;
  exports.isSvg = isSvg;
  exports.noop = noop;

  Object.defineProperty(exports, '__esModule', { value: true });

}));
